### PR TITLE
Solution: Sprint Challenge -- Applied JavaScript

### PR DESCRIPTION
**Pull request by nickcannariato**
## Minimum Viable Product

Your finished project must include all of the following requirements:

- [x] Look through the HTML code paying particular attention to the Tabs component and the Cards components. You will notice they share a data attribute. We will be using this data attribute to determine which cards should show when each tab is selected.

- [x] Following the instructions in the `Tabs.js` file, complete the `TabLink`, and `TabCard` class components. It will look and feel very similar to the last project we worked on, but with a twist. Now, instead of one `Item` to display, we will need to display a collection of `Cards`. Think about ways to iterate over an array and manipulate each item. **Note: You will need to un-comment the code after the lines of instructions. The code is commented out so you can work error-free**

- [x] Once you get your `Tab` component working properly add a couple more articles yourself and check out how it works.

## Stretch Problems

Your stretch challenge is to write the functionality of a `Carousel` component. You have the HTML and CSS already in place, simply un-comment the HTML in the `index.html` file. This is an advanced challenge, so you are not expected to be able to complete it. If you begin and don't finish, you should still submit with what you have. You may reference the `Tabs.js` file for assistance.

- [x] Complete the carousel functionality in `Carousel.js`

- [x] If you complete the Carousel, add functionality so that the carousel slides when the buttons are pressed instead of just appearing.

- [x] Create an 'infinite loop' carousel. In which as long as you click on an arrow, the array of images will loop over itself.

- [x] If you have finished the above, play around with the styling on all the components, and understand how each is built.

### CODE REVIEW

File: CSS/index.css [hunk 1]
```diff
@@ -300,5 +301,34 @@ body {
...
+}
+.carousel .fade-in {
+  -webkit-animation: fade-in 0.5s cubic-bezier(0.39, 0.575, 0.565, 1) both;
+  animation: fade-in 0.5s cubic-bezier(0.39, 0.575, 0.565, 1) both;
+}
+/**
+ * ----------------------------------------
+ * animation fade-in
+ * ----------------------------------------
+ */
```
Comment (sabrinagear) [inline CSS/index.css]: Love the animation on the carousel. Makes for a much nicer transition.

File: README.md [hunk 2]
```diff
@@ -32,39 +32,49 @@ Edit this document to include your answers after each question. Make sure to lea
 
 1. What is the DOM?
 
+> The DOM is the Document Object Model, and it is an API for HTML and XML documents that represents the page so that programs can change the document structure, style, and content
+
```
Comment (sabrinagear) [inline README.md]: Good.

File: README.md [hunk 3]
```diff
@@ -32,39 +32,49 @@ Edit this document to include your answers after each question. Make sure to lea
...
+> An event listener is a higher order function that allows you to specify a DOM event to listen for and a function to execute when that event is fired.
+
 4. Why would we convert a NodeList into an Array?
 
+> Because NodeLists, while Array-like, are not Arrays and don't have access to many of the methods provided by the Array.
+
 5. What is a component?
 
+> Components are “building blocks” of web applications and pages. Specifically they’re reusable blocks of HTML, CSS, and JS.
+
```
Comment (sabrinagear) [inline README.md]: Nice characterization.

File: components/Carousel/Carousel.js [hunk 4]
```diff
@@ -1,14 +1,34 @@
...
+            image.classList.remove('fade-in');
+        });
+
+        if (e.target === this.leftButton) {
+            this.currentIndex < 1 ? this.currentIndex = maxIndex : this.currentIndex--;
+        } else {
+            this.currentIndex === maxIndex ? this.currentIndex = 0 : this.currentIndex++;
+        }
+        this.images[this.currentIndex].classList.add('fade-in')
+    }
```
Comment (sabrinagear) [inline components/Carousel/Carousel.js]: Great job, Nick!

File: components/Carousel/Carousel.less [hunk 5]
```diff
@@ -18,24 +19,56 @@
...
+  0% {
+    opacity: 0;
+  }
+  100% {
+    opacity: 1;
+  }
+}
+@keyframes fade-in {
+  0% {
+    opacity: 0;
```
Comment (sabrinagear) [inline components/Carousel/Carousel.less]: Nice job! The carousel is going over the nav-bar though. Can you fix that?

Comment (nickcannariato) [reply components/Carousel/Carousel.less]: Ah! I knew I missed something on the z-index adjustment 🤦‍♂️ 

I'll fix it in master since this is merged. 😄

Comment (nickcannariato) [reply components/Carousel/Carousel.less]: @sabrinagear Issue fixed in f5666f8 👍

**Pull request merged**